website/www/site/content/en/documentation/patterns/state-and-timers.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "State and timers patterns"
+---
+
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# State and timers for calling external services
+
+Usually, authoring an Apache Beam pipeline can be done with out of the box tools and transforms like _ParDo_'s, _Window_'s and _GroupByKey_'s. However, when you want more tight control, keeping state in an otherwise stateless _DoFn_.
+
+State is kept on a per-key (and in case of streaming pipelines, per-windows basis), and as such, the input to your stateful DoFn needs to be keyed (e.g. by the customer identifier is you're tracking clicks from an e-commerce website).
```

Comment:
Its still per window even in batch ( just happens to be global window most of the time but not always)